src/plots.jl

```diff
@@ -418,4 +419,96 @@ function Histogram2(
     Patch2D(patchdata, style=style, patch_type="rectangle")
 end
 
+mutable struct MatrixPlot <: Plot
+    filename::AbstractString
+    xmin::Real
+    xmax::Real
+    ymin::Real
+    ymax::Real
+    zmin::Real
+    zmax::Real
+	rows::Real
+	cols::Real
+    zmode
+	raster
+    colorbar::Bool
+    colorbarStyle
+    colormap::ColorMaps.ColorMap
+    style
+    function MatrixPlot(
+        A::AbstractMatrix{T},
+        xrange::Union{Nothing,RealRange}=nothing,
+        yrange::Union{Nothing,RealRange}=nothing;
+        filename=nothing,
+        colorbar=true,
+        colorbarStyle=nothing,
+        colormap=ColorMaps.GrayMap(),
+        zmin=nothing,
+        zmax=nothing,
+        zmode=nothing,
+		raster=nothing,
+        style=nothing,
+        ) where {T <: Real}
+		
+		global _imgid
+        if filename == nothing
+            id=myid()*10000000000000+_imgid
+            filename = "tmp_$(id).dat"
+            _imgid += 1
+        end
+		
+		(rows,cols) = size(A)
+		if xrange == nothing
+			xrange = (1,cols)
+		end	
+		if yrange == nothing
+			yrange = (1,rows)
+		end
+        if zmin == nothing
+            zmin = minimum(A[.!isnan.(A)])
+        end
+        if zmax == nothing
+            zmax = maximum(A[.!isnan.(A)])
+        end
+        if zmin == zmax
+            zmin -= 1.
+            zmax += 1.
+        end
+
+		if length(A) <= 1e4
+			raster = false
+		elseif raster == nothing
+			@warn "Matrix is too large for vector plotting, consider switching to raster mode."
+			raster = true			
+		end
+		
+		if raster
+			file, _ = splitext(filename)
+			filename = string(file,".png")
+			A = clamp.(A, zmin, zmax)
+			A = A .- zmin
+			A = A ./ (zmax - zmin)
+			if isa(colormap, ColorMaps.ColorMap)
+				write(colormap, A, filename)
+			else
+				write(ColorMaps.RGBArrayMap(colormap), A, filename)
+			end
+		else
+			out = Array{Any}(undef,length(A)+1,3)
+			x = range(xrange[1], stop=xrange[2], length=cols)
+			y = range(yrange[1], stop=yrange[2], length=rows)
+			x = x .- 0.5*step(x)
```

Comment:
Slight preference for
```
x .-= step(x)/2
y .-= step(y)/2
```

Reply:
So I tried this initially, but I get an error
```julia
ERROR: setindex! not defined for StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}```.
For the time being I'll leave it as is, but if there's something I missed in the implementation I can go back and fix it.